presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveDynamicPartitionPruning.java

```diff
@@ -0,0 +1,240 @@
+                getSession(),
+                "SELECT * FROM partitioned_lineitem JOIN orders ON partitioned_lineitem.orderkey = orders.orderkey");
+        assertEquals(result.getResult().getRowCount(), LINEITEM_COUNT);
+
+        OperatorStats probeStats = searchScanFilterAndProjectOperatorStats(result.getQueryId(), "tpch:" + PARTITIONED_LINEITEM);
+        // Probe-side is fully scanned because the build-side is too large for dynamic filtering:
+        assertEquals(probeStats.getInputPositions(), LINEITEM_COUNT);
+        assertEquals(probeStats.getDynamicFilterSplitsProcessed(), 0L);
+
+        DynamicFiltersStats dynamicFiltersStats = getDynamicFilteringStats(result.getQueryId());
+        assertEquals(dynamicFiltersStats.getTotalDynamicFilters(), 1L);
+        assertEquals(dynamicFiltersStats.getLazyDynamicFilters(), 1L);
+        assertEquals(dynamicFiltersStats.getReplicatedDynamicFilters(), 0L);
+        assertEquals(dynamicFiltersStats.getDynamicFiltersCompleted(), 1L);
+
+        DynamicFilterDomainStats domainStats = getOnlyElement(dynamicFiltersStats.getDynamicFilterDomainStats());
+        assertEquals(domainStats.getSimplifiedDomain(), Domain.all(INTEGER).toString(getSession().toConnectorSession()));
+        assertEquals(domainStats.getDiscreteValuesCount(), 0);
+        assertEquals(domainStats.getRangeCount(), 1);
+    }
+
+    @Test(timeOut = 30_000)
+    public void testJoinWithMultipleDynamicFiltersOnProbe()
+    {
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        // supplier names Supplier#000000001 and Supplier#000000002 match suppkey 1 and 2
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                getSession(),
+                "SELECT * FROM (" +
+                        "SELECT supplier.suppkey FROM " +
+                        "partitioned_lineitem JOIN tpch.tiny.supplier ON partitioned_lineitem.suppkey = supplier.suppkey AND supplier.name IN ('Supplier#000000001', 'Supplier#000000002')" +
+                        ") t JOIN supplier ON t.suppkey = supplier.suppkey AND supplier.suppkey IN (2, 3)");
+        assertGreaterThan(result.getResult().getRowCount(), 0);
+
+        OperatorStats probeStats = searchScanFilterAndProjectOperatorStats(result.getQueryId(), "tpch:" + PARTITIONED_LINEITEM);
+        // Probe-side is partially scanned
+        assertBetweenInclusive(probeStats.getInputPositions(), 558L, LINEITEM_COUNT - 1);
+        assertEquals(probeStats.getDynamicFilterSplitsProcessed(), 0L);
+
+        DynamicFiltersStats dynamicFiltersStats = getDynamicFilteringStats(result.getQueryId());
+        assertEquals(dynamicFiltersStats.getTotalDynamicFilters(), 2L);
+        assertEquals(dynamicFiltersStats.getLazyDynamicFilters(), 2L);
+        assertEquals(dynamicFiltersStats.getReplicatedDynamicFilters(), 0L);
+        assertBetweenInclusive(dynamicFiltersStats.getDynamicFiltersCompleted(), 1, 2);
```

Comment:
@sopel39 FYI I tweaked this test a bit to assert on completion of only one filter. I was seeing some flakiness without this because the probe side gets unblocked after receiving one filter and the query can sometimes finish before DynamicFilterService is able to collect the 2nd filter. If I set `experimental.dynamic-filtering-refresh-interval` to a very low value like `20ms` then the flakiness goes away.

Reply:
After changing to wait on full DF, the above tweak is no longer needed